content/learn/book/welcome/plugins/_index.md

```diff
@@ -1,16 +1,107 @@
 +++
-title = "Modularity through plugins"
-weight = 4
+title = "Plugins"
+weight = 3
 template = "book-section.html"
 page_template = "book-section.html"
 +++
 
-TODO: Explain how plugins work
+One of Bevy's core principles is modularity. In Bevy, all functionality is implemented via {{rust_type(type="trait" crate="bevy_app" name="Plugin" plural=true)}}, which are added to an {{rust_type(type="struct" crate="bevy_app", name="App")}}. Game logic like player movement, core engine logic like rendering and sound, and third party extensions like tile maps are all implemented the same way using {{rust_type(type="trait" crate="bevy_app" name="Plugin" plural=true)}}.
+
+This empowers Bevy developers to modularly "build their own engine" using official, third party, and custom {{rust_type(type="trait" crate="bevy_app" name="Plugin" plural=true)}}. Bevy intentionally blurs the lines between engine developers and app developers.
+
+## Writing your own plugins
+
+Plugins are collections of code that modify {{rust_type(type="trait" crate="bevy_app" name="App" plural=true)}}.
+Any code in a plugin could be directly substituted directly on the base `App`.
+There's no magic to be found here; they're just a straightforward tool for code organization.
+
+Plugins are types that implement the {{rust_type(type="trait" crate="bevy_app" name="Plugin")}} trait:
+
+```rust
+use bevy::prelude::*;
+
+fn main(){
+  App::build()
+    // Adds the "default" Bevy Engine plugins.
+    // We'll cover this in the next section.
+    .add_plugins(DefaultPlugins)
+    // Adds our new `Plugin` to the `App`
+    .add_plugin(ScorePlugin)
+    .run();
+}
+
+struct ScorePlugin;
+
+impl Plugin for ScorePlugin {
+  fn build(&self, app: &mut App) {
+    app
+      // The Score struct is added as a resource (global singleton) to the world, 
+      // beginning at the default value of 0
+      .init_resource::<Score>()
+      // Increments the score by 1 every pass of the game loop
+      .add_system(increment_score.system())
+      // Prints the current value of the score
+      .add_system(report_score.system());
+  }
+}
+
+#[derive(Default, Debug)]
+struct Score(u8);
+
+fn increment_score(score: ResMut<Score>){
+  score.0 += 1;
+}
+
+fn report_score(score: Res<Score>){
+  info!(score);
+}
+```
+
+## `DefaultPlugins` and `PluginGroup`
```

Comment:
Monospace text is rendered particularly small in headings. Also, the paragraph should talk about plugin groups in general (with an appropriate subtitle) and then introduce both `DefaultPlugins` and `MinimalPlugins`.

![lol](https://user-images.githubusercontent.com/32368687/128601392-982df4eb-aec2-40d3-8425-b83ce19c4cef.png)

Reply:
I called the formatting problem out here too: https://github.com/bevyengine/bevy-website/pull/176#discussion_r661047979
We either need to:
1. fix "code formatted headings" to be the right size
2. remove code formatting in headings

I'm actually sort of in favor of (2) because code blocks in headings feel kind of "noisy" to me. But I'm happy to try (1) and see how we feel.

I don't think this needs to be resolved in this pr, but it needs to be resolved prior to "releasing" the new version of the book.

Reply:
Obviously (1) must be done sooner or later, but we really shouldn't abuse mentioning identifiers in the Book, especially in headings. I'm on the opinion that we should not strictly follow the structure of the API (otherwise we get a bad copy of the API docs), but that thoughts should be laid out in a way that give access to a comprehension level that reading source code or API docs can't do.

For the header, in this case, "Plugin groups" should be sufficient.